pkg/controller/jobs/job/job_controller.go

```diff
@@ -247,7 +263,17 @@ func (j *Job) RestorePodSetsInfo(podSetsInfo []podset.PodSetInfo) bool {
 			j.Spec.Completions = j.Spec.Parallelism
 		}
 	}
-	changed = podset.RestorePodSpec(&j.Spec.Template.ObjectMeta, &j.Spec.Template.Spec, podSetsInfo[0]) || changed
+	info := podSetsInfo[0]
+	for _, mangedLabel := range mangedLabels {
+		if v, found := j.Spec.Template.Labels[mangedLabel]; found {
+			if info.Labels == nil {
```

Comment:
I'm wondering what we could do to avoid checks like this scattered in codebase. I would suggest either:
1. make sure `info.Labels` is not nil  when creating in `FromPodSet`, as we do in the `FromAssignment` already, so seems to make sense being consistent: https://github.com/kubernetes-sigs/kueue/blob/615dc8e42d160174962f379dd8767e95298a1457/pkg/podset/podset.go#L56
2. use some utility functions to merge the maps for the managedLabels. I think creating an extra map to use `MergeKeepFirst` is fine, but if we want to avoid that, maybe we could have a new utility function `MergeInto`.

Reply:
Also we could add a method on PodSetInfo lile AddOrUpdateLabel(k,v)...

Reply:
going with the `AddOrUpdate` or `AddOrUpdateLabel` sgtm

Reply:
AddOrUpdateLabel done